### DISCUSSION TITLE
Entry versions don't always sync with entry saves

### DISCUSSION THREAD

**Discussion by robinsowell** [ExpressionEngine/ExpressionEngine#3677] [Feature Requests]
### Suggest an idea for ExpressionEngine

When you save an entry and have versioning enabled, you expect it to make a corresponding entry version entry.  BUT if you save an entry without changing anything, it does not make an entry version entry.   It DOES however update the edit date on the entry (exp_channel_title table).  We know this because the user was really confused that the edit dates for the entry didn't all have a corresponding entry version.

I do get that saving duplicate data as a version is not the most optimized behavior but I also get that if you have an entry edit, no matter what the edit you expect it to have a version.

Should we add the duplicate entry version?  Should we not, and document it?  I definitely think we SHOULD keep updating the entry edit_date.  So.  Discussion!

**Comment by swierczek** [comment]
For reference, https://github.com/ExpressionEngine/ExpressionEngine/blob/7.dev/system/ee/ExpressionEngine/Model/Channel/ChannelEntry.php#L648 is what causes this behavior.

I think the entry version should be saved regardless of its content, because the user saved the entry and it's slightly confusing UX to have the revisions not align directly with the saves. My recommendation would be to probably not change the current behavior, but to instead add a config value like `$config['save_identical_revisions'] = true;` (with a default value of `false`) that would be checked in the conditional in ChannelEntry.php to allow site developers to customize this behavior as needed.

**Comment by intoeetive** [comment]
I think the reason for this behavior was to prevent versions table in database grow too big without strong reasons, and also make sure saved revisions (as by default we keep only latest 10)  are actually useful. But I agree, having a setting here should be better
